extensions/justwatch-search/src/api.ts

```diff
@@ -0,0 +1,93 @@
+import fetch from "node-fetch";
+import { LocalStorage, showToast, Toast } from "@raycast/api";
+import { JustWatchMedia, JustWatchMediaOffers, MediaProvider } from "./types";
+
+export async function searchMedias(query: string): Promise<JustWatchMedia[]> {
+  const countryCode = (await LocalStorage.getItem<string>("country_code")) || "en_CA";
+
+  const url = `https://apis.justwatch.com/content/titles/${countryCode}/popular?language=en&body={"page_size":8,"page":1,"query":"${query}","content_types":["show","movie"]}`;
```

Comment:
Hm, this is buggy to url manipulation. For example, if someone types in the search bar a quote e.g: `example " here` then this query will break because the body will include an invalid JSON payload:
```
body={"page_size":8,"page":1,"query":"example" here","content_types":["show","movie"]}
```
☝️ see this is invalid `"query":"example" here"`. Using plain string concatenation from user input for building URLs isn't a good practice.

You can use the `URLSearchParams` constructor instead to ensure all query params are encoded properly and set `body` as stringified JSON object:
```ts
const searchParams = new URLSearchParams({
  language: "en",
  body: JSON.stringify({
    page_size: 8,
    page: 1,
    query,
    content_types: ["show", "movie"]
  })
})
const url = new URL(`/content/titles/${countryCode}/popular?${searchParams}`, 'https://apis.justwatch.com').toString()
// or (but prefer the one above)
// const url = `https://apis.justwatch.com/content/titles/${countryCode}/popular?${searchParams}`
```

Reply:
Good catch, thank you! I used your suggestion for `URLSearchParams` and pushed.